snapshot/android/project-snapshot-generator.js

```diff
@@ -178,23 +182,29 @@ ProjectSnapshotGenerator.prototype.generate = function(generationOptions) {
 
     // Generate snapshots
     const generator = new SnapshotGenerator({ buildPath: this.getBuildPath() });
-    return generator.generate({
-        snapshotToolsPath,
-        inputFile: generationOptions.inputFile || join(this.options.projectRoot, "__snapshot.js"),
-        targetArchs: generationOptions.targetArchs || ["arm", "arm64", "ia32"],
-        v8Version: generationOptions.v8Version || this.getV8Version(),
-        preprocessedInputFile: generationOptions.preprocessedInputFile,
-        useLibs: generationOptions.useLibs || false,
-        androidNdkPath
-    }).then(() => {
-        console.log("Snapshots build finished succesfully!");
-
-        if (generationOptions.install) {
-            ProjectSnapshotGenerator.cleanSnapshotArtefacts(this.options.projectRoot);
-            ProjectSnapshotGenerator.installSnapshotArtefacts(this.options.projectRoot);
-            console.log(generationOptions.useLibs ? 
-                "Snapshot is included in the app as dynamically linked library (.so file)." :
-                "Snapshot is included in the app as binary .blob file. The more space-efficient option is to embed it in a dynamically linked library (.so file).");
-        }
+
+    return this.getV8Version(generationOptions).then(v8Version => {
+        return generator.generate({
+            snapshotToolsPath,
+            inputFile: generationOptions.inputFile || join(this.options.projectRoot, "__snapshot.js"),
+            targetArchs: generationOptions.targetArchs || ["arm", "arm64", "ia32"],
+            v8Version: generationOptions.v8Version || this.getV8Version(),
+            preprocessedInputFile: generationOptions.preprocessedInputFile,
+            useLibs: generationOptions.useLibs || false,
+            androidNdkPath
+        }).then(() => {
+            console.log("Snapshots build finished succesfully!");
+
+            if (generationOptions.install) {
+                ProjectSnapshotGenerator.cleanSnapshotArtefacts(this.options.projectRoot);
+                ProjectSnapshotGenerator.installSnapshotArtefacts(this.options.projectRoot);
+                console.log(generationOptions.useLibs ?
+                    "Snapshot is included in the app as dynamically linked library (.so file)." :
+                    "Snapshot is included in the app as binary .blob file. The more space-efficient option is to embed it in a dynamically linked library (.so file).");
+            }
+        });
+    }).catch(error => {
+        throw new Error("Cannot find suitable v8 version!");
```

Comment:
Do you think it would be sensible to add information about the actual error that occurred here, like for example:
`` throw new Error(`Cannot find suitable v8 version! Original error: ${error.message || error}`); ``